lib/console.js

```diff
@@ -249,6 +264,108 @@ Console.prototype.groupEnd = function groupEnd() {
+const ITKEY = '(iteration index)';
+
+const inspect = (v) => {
+  const opt = { depth: 0, maxArrayLength: 3 };
+  if (v !== null && typeof v === 'object' &&
+      !ArrayIsArray(v) && ObjectKeys(v).length > 2)
+    opt.depth = -1;
+  return util.inspect(v, opt);
+};
+
+const increasingArray = (n) => ArrayFrom({ length: n }, (_, i) => inspect(i));
+
+// https://console.spec.whatwg.org/#table
+Console.prototype.table = function(tabularData, properties) {
+  if (tabularData == null ||
+      ['symbol', 'boolean', 'number', 'string'].includes(typeof tabularData) ||
+      (ArrayIsArray(tabularData) && tabularData.length > 20) ||
+      ObjectKeys(tabularData) > 20)
+    return this.log(tabularData);
+
+  const mapIter = isMapIterator(tabularData);
+  let setIter;
+  if (mapIter)
+    tabularData = previewMapIterator(tabularData);
+  else if (setIter = isSetIterator(tabularData))
+    tabularData = previewSetIterator(tabularData);
+
+  const final = (k, v) => this.log(cliTable(k, v));
+
+  if (mapIter || isMap(tabularData)) {
+    const kkeyArr = [];
+    const vkeyArr = [];
+    for (const [k, v] of tabularData) {
+      kkeyArr.push(inspect(k));
+      vkeyArr.push(inspect(v));
+    }
+    return final([
+      ITKEY, KKEY, VKEY
+    ], [
+      increasingArray(tabularData.length || tabularData.size),
+      kkeyArr,
+      vkeyArr,
+    ]);
+  } else if (setIter || isSet(tabularData)) {
+    const vkeyArr = [];
+    for (const v of tabularData)
+      vkeyArr.push(inspect(v));
+    return final([ITKEY, VKEY], [
+      increasingArray(tabularData.length || tabularData.size),
+      vkeyArr,
+    ]);
+  } else if (typeof tabularData === 'object' && !ArrayIsArray(tabularData)) {
+    const map = {};
+    const keys = properties || Object.keys(tabularData);
+    for (const key of keys)
+      map[key] = [inspect(tabularData[key])];
+    return final(ObjectKeys(map), ObjectValues(map));
+  } else {
+    const ikeyArr = [];
+    const vkeyArr = [];
+    let hasPrimitives = false;
+    const map = {};
+    for (var i = 0; i < tabularData.length; i++) {
+      ikeyArr.push(inspect(i));
+      const item = tabularData[i];
+      if (properties !== undefined) {
+        for (const key of properties) {
+          if (map[key] === undefined)
+            map[key] = [];
+          map[key][i] = inspect(item[key]);
+        }
+      } else if (item === null ||
+                 (typeof item !== 'function' && typeof item !== 'object')) {
+        hasPrimitives = true;
+        vkeyArr[i] = inspect(item);
+      } else {
+        const descriptors = ObjectGetOwnPropertyDescriptors(item);
+        for (const [key, desc] of ObjectEntries(descriptors)) {
+          const has = map[key] !== undefined;
+          if (!has && !desc.enumerable)
+            continue;
+          if (!has)
+            map[key] = [];
+          map[key][i] = inspect(item[key]);
+        }
+      }
+    }
+    const keys = ObjectKeys(map);
+    const values = ObjectValues(map);
```

Comment:
This is unnecessary overhead and could be simplified by directly saving the keys and values in a dedicated array.

Reply:
i think this is probably faster than
a) iterating all the objects first and building an array of keys, then iterating again and plucking values
b) one loop but performing array index lookup for each key

if there's an option c i'd be glad to hear it :)